### PR TITLE
Backup and restore orignal LIBPATH on z/OS Java 21+

### PR DESCRIPTION
**Pull request by hangshao0**
Backup and restore orignal LIBPATH on z/OS Java 21+

related to https://github.com/eclipse-openj9/openj9/pull/20007

### CODE REVIEW

**Comment by hangshao0** [comment]
Internal [build](https://hyc-runtimes-jenkins.swg-devops.com/job/Build_JDK21_s390x_zos_Personal/1377/) to test this change.

**Comment by hangshao0** [comment]
There are more change needed, converting this to draft.

**Comment by hangshao0** [comment]
Internal [build ](https://hyc-runtimes-jenkins.swg-devops.com/job/Build_JDK21_s390x_zos_Personal/1383/) passed.

**Comment by pshipton** [comment]
Pls update the commit and PR description to describe what the fix is related to, i.e. 
Backup/restore LIBPATH on z/OS Java 21+

**Comment by hangshao0** [comment]
> Pls update the commit and PR description to describe what the fix is related to, i.e.
Backup/restore LIBPATH on z/OS Java 21+

Done.

File: runtime/j9vm/jvm.c [hunk 1]
```diff
@@ -3068,9 +3065,6 @@ restoreLibpath(J9LibpathBackup *libpathBackup)
 		libpathBackup->j9prefixLen = 0;
 	}
 }
-#endif /* AIXPPC */
-
-#if defined(AIXPPC)
```
Comment (keithc-ca) [inline runtime/j9vm/jvm.c:3073]: Please leave the blank line between functions.

File: runtime/j9vm/jvm.c [hunk 2]
```diff
@@ -3184,9 +3176,9 @@ deleteDirsFromLibpath(const char *const libpath, const char *const deleteStart,
 
 	return newPath;
 }
-#endif /* AIXPPC */
+#endif /* defined(AIXPPC) || (defined(J9ZOS390) && (JAVA_SPEC_VERSION >= 21)) */
 
-#if defined(AIXPPC) && defined(DEBUG_TEST)
+#if (defined(AIXPPC) || (defined(J9ZOS390) && (JAVA_SPEC_VERSION >= 21))) && defined(DEBUG_TEST)
```
Comment (keithc-ca) [inline runtime/j9vm/jvm.c:3181]: Please correct the comment in the matching `#endif`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: runtime/j9vm/jvm.c (hunk 1 above)
Comment (hangshao0) [reply runtime/j9vm/jvm.c]: Fixed.

File: runtime/j9vm/jvm.c (hunk 2 above)
Comment (hangshao0) [reply runtime/j9vm/jvm.c:3182]: Fixed.

**Review by keithc-ca — APPROVED** [review]

**Comment by hangshao0** [comment]
I am going to revert this change (in a new PR). The code to backup and restore LIBPATH is only to remove the prepended j9bin and jrebin in preloadLibraries() on AIX:

https://github.com/eclipse-openj9/openj9/blob/91b5703ea59e75ae6c72e4c65123ba13bc06488a/runtime/j9vm/jvm.c#L1319
https://github.com/eclipse-openj9/openj9/blob/91b5703ea59e75ae6c72e4c65123ba13bc06488a/runtime/j9vm/jvm.c#L1392-L1395

The appended `usr/lib` on AIX still stays in LIBPATH. https://github.com/eclipse-openj9/openj9/pull/20007 is to always add `/lib` to LIBPATH on z/OS.

**Comment by hangshao0** [comment]
> I am going to revert this change (in a new PR)

Here: https://github.com/eclipse-openj9/openj9/pull/20070

**Pull request merged**